internal/service/api/handlers/register.go

```diff
@@ -51,29 +51,17 @@ func Register(w http.ResponseWriter, r *http.Request) {
 	documentSOD := data.DocumentSOD{
 		HashAlgorigthm:      algorithmPair.HashAlgorithm,
 		SignatureAlgorithm:  algorithmPair.SignatureAlgorithm,
-		SignedAttributes:    utils.TruncateHexPrefix(req.Data.Attributes.DocumentSod.SignedAttributes),
-		EncapsulatedContent: utils.TruncateHexPrefix(req.Data.Attributes.DocumentSod.EncapsulatedContent),
-		Signature:           utils.TruncateHexPrefix(req.Data.Attributes.DocumentSod.Signature),
+		SignedAttributes:    *utils.TruncateHexPrefix(&req.Data.Attributes.DocumentSod.SignedAttributes),
+		EncapsulatedContent: *utils.TruncateHexPrefix(&req.Data.Attributes.DocumentSod.EncapsulatedContent),
+		Signature:           *utils.TruncateHexPrefix(&req.Data.Attributes.DocumentSod.Signature),
```

Comment:
Are you sure there is no panic due to nil pointer dereference in case when `nil` returned and you're trying to dereference result?